src/core/server/services/slack/publisher.ts

```diff
@@ -0,0 +1,214 @@
+  const p: any = payload;
+  return (
+    channel === SUBSCRIPTION_CHANNELS.COMMENT_ENTERED_MODERATION_QUEUE &&
+    p.hasOwnProperty("queue") &&
+    (p.queue === GQLMODERATION_QUEUE.REPORTED ||
+      p.queue === GQLMODERATION_QUEUE.PENDING)
+  );
+}
+
+function isFeatured(channel: SUBSCRIPTION_CHANNELS) {
+  return channel === SUBSCRIPTION_CHANNELS.COMMENT_FEATURED;
+}
+
+function isReported(channel: SUBSCRIPTION_CHANNELS, payload: Payload) {
+  const p: any = payload;
+  return (
+    channel === SUBSCRIPTION_CHANNELS.COMMENT_ENTERED_MODERATION_QUEUE &&
+    p.hasOwnProperty("queue") &&
+    p.queue === GQLMODERATION_QUEUE.REPORTED
+  );
+}
+
+function isPending(channel: SUBSCRIPTION_CHANNELS, payload: Payload) {
+  const p: any = payload;
+  return (
+    channel === SUBSCRIPTION_CHANNELS.COMMENT_ENTERED_MODERATION_QUEUE &&
+    p.hasOwnProperty("queue") &&
+    p.queue === GQLMODERATION_QUEUE.PENDING
+  );
+}
+
+function createModerationLink(rootURL: string, commentID: string) {
+  return `${rootURL}/admin/moderate/comment/${commentID}`;
+}
+
+function createCommentLink(storyURL: string, commentID: string) {
+  return `${storyURL}?commentID=${commentID}`;
+}
+
+async function postCommentToSlack(
+  ctx: SlackContext,
+  orgURL: string,
+  commentID: string,
+  webhookURL: string
+) {
+  const comment = await ctx.comments.load(commentID);
+  if (comment === null) {
+    return;
+  }
+  const story = await ctx.stories.load(comment.storyID);
+  if (story === null) {
+    return;
+  }
+  const author = await ctx.users.load(comment.authorID);
+  if (author === null) {
+    return;
+  }
+
+  const storyTitle = story.metadata ? story.metadata.title : "";
+  const commentBody =
+    comment.revisions.length > 0
+      ? comment.revisions[comment.revisions.length - 1].body
+      : "";
+  const body = commentBody.replace(new RegExp("<br>", "g"), "\n");
+  const moderateLink = createModerationLink(orgURL, comment.id);
+  const commentLink = createCommentLink(story.url, comment.id);
+
+  const data = {
+    text: `${author.username} commented on: ${storyTitle}`,
+    blocks: [
+      {
+        type: "section",
+        text: {
+          type: "mrkdwn",
+          text: `${author.username} commented on:\n<${story.url}|${storyTitle}>`,
+        },
+      },
+      {
+        type: "section",
+        text: {
+          type: "mrkdwn",
+          text: `<${moderateLink}|Go to Moderation> | <${commentLink}|See Comment>`,
+        },
+      },
+      {
+        type: "divider",
+      },
+      {
+        type: "section",
+        text: {
+          type: "mrkdwn",
+          text: body,
+        },
+      },
+    ],
+  };
+
+  try {
+    const response = await fetch(webhookURL, {
+      method: "POST",
+      mode: "cors",
+      cache: "no-cache",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      redirect: "follow",
+      referrer: "no-referrer",
+      body: JSON.stringify(data),
+    });
```

Comment:
Why is `cors` specified? This shouldn't apply server side. Also is there a reason `redirect` and `referrer` are set?